sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -148,6 +149,10 @@ void dispose(ErrorCondition errorCondition) {
 
         openReceiveLinks.forEach((key, link) -> link.dispose(errorCondition));
         openSendLinks.forEach((key, link) -> link.dispose(errorCondition));
+
+        if (transactionCoordinator.get() != null) {
```

Comment:
`get()` is atomic. It's possible when you get it a second time, it'll be null on .dispose(). The usual is to assign it to a local variable.